diff -up QR-Code-generator-1.5.0/python/qrcodegen-batch-test.py.1~ QR-Code-generator-1.5.0/python/qrcodegen-batch-test.py
--- QR-Code-generator-1.5.0/python/qrcodegen-batch-test.py.1~	2019-07-16 01:16:50.000000000 +0200
+++ QR-Code-generator-1.5.0/python/qrcodegen-batch-test.py	2019-12-17 21:27:41.055157452 +0100
@@ -25,6 +25,8 @@
 # 
 
 import itertools, random, subprocess, sys, time
+import os
+import os.path
 if sys.version_info.major < 3:
 	raise RuntimeError("Requires Python 3+")
 
@@ -47,8 +49,14 @@ def main():
 	global subprocs
 	try:
 		for args in CHILD_PROGRAMS:
+			env = dict(os.environ)
+			if args[0].startswith('..'):
+				ld_path = os.path.dirname(args[0])
+				if 'LD_LIBRARY_PATH' in env:
+					ld_path += ':' + env['LD_LIBRARY_PATH']
+				env['LD_LIBRARY_PATH'] = ld_path
 			subprocs.append(subprocess.Popen(args, universal_newlines=True,
-				stdin=subprocess.PIPE, stdout=subprocess.PIPE))
+				stdin=subprocess.PIPE, stdout=subprocess.PIPE, env=env))
 	except FileNotFoundError:
 		write_all(-1)
 		raise
@@ -63,7 +71,7 @@ def main():
 		sys.exit("Error: One or more workers failed to start")
 	
 	# Do tests
-	for i in itertools.count():
+	for i in range(50):
 		print("Trial {}: ".format(i), end="")
 		do_trial()
 		print()
